### ISSUE TITLE
tinytga crate panics with (some) rle compressed images

### ISSUE THREAD

**Issue by david-sawatzke** [jamwaffles/embedded-graphics#216]
- Version of embedded-graphics in use (if applicable): git master (4548c) and 0.6.0-alpha2
 (tinytga crate version 0.1.0)
- Target: both

## Description of the problem/feature request/other

When the xcf in the attached tar is exported with RLE compression to tga and loaded in the tga example, it panics (the tga file is also attached). Without compression, it works fine. 

## Test case (if applicable)
Replace `rust-pride.tga` with the attached tga and then run it:
```
$ cargo run --example tga
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `[..]/embedded-graphics/target/debug/examples/tga`
thread 'main' panicked at 'attempt to multiply with overflow', /[..]/.cargo/registry/src/github.com-1ecc6299db9ec823/tinytga-0.1.0/src/packet/raw_pack
et.rs:29:27                                                                                                                                                 
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

$ cargo run --example tga --release
    Finished release [optimized] target(s) in 0.04s
     Running `[..]/embedded-graphics/target/release/examples/tga`
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', [..]/.cargo/registry/src/github.com-1ecc6299db9ec823/tinytga-0.1.0/src/lib.rs:226:42
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
[reproduce.tar.gz](https://github.com/jamwaffles/embedded-graphics/files/3965497/reproduce.tar.gz)

**Comment by jamwaffles** [comment]
Thanks for the bug report! I have to admit the RLE code hasn't been tested particularly well. I'll get round to investigating a fix when I have some spare time.

**Comment by rfuest** [comment]
While working on the `image` module I also did some work on `tinytga`, like a nom 5 port and adding additional test images. Because this was outside the scope of that PR I didn't commit the changes back then and forgot about theme afterwards. Maybe these changes would also influence this bug.

**Comment by jamwaffles** [comment]
Both `tinytga` and `tinybmp` could use some polish and updates so even if your changes don't resolve this issue it would be good to get them committed.

**Comment by jamwaffles** [comment]
This is fixed by the awesome @rfuest in version [0.2.0](https://crates.io/crates/tinytga/0.2.0) of `tinytga`. Please reopen this issue if you continue to have problems. Thanks!

**Comment by david-sawatzke** [comment]
Thanks @rfuest, the image works fine now
